app/controllers/topics_controller.rb

```diff
@@ -9,12 +9,19 @@ class TopicsController < ApplicationController
   # GET /topics/autocomplete_topic
   def autocomplete_topic
     term = params[:term]
-    @topics = if term && !term.empty?
-                policy_scope(Topic.where('LOWER("name") like ?', term.downcase + '%')).order('"name"')
-              else
-                []
-              end
-    render json: autocomplete_array_json(@topics)
+    if term && !term.empty?
+      @topics = policy_scope(Topic.where('LOWER("name") like ?', term.downcase + '%')).order('"name"')
+      @mapTopics = @topics.select { |t| t.metacode.name == 'Metamap' }
+      # prioritize topics which point to maps, over maps
+      @exclude = @mapTopics.length > 0 ? @mapTopics.map(&:name) : ['']
+      @maps = policy_scope(Map.where('LOWER("name") like ? AND name NOT IN (?)', term.downcase + '%', @exclude)).order('"name"')
+    else
+      @topics = []
+      @maps = []
+    end
+    @all= @topics.concat(@maps).sort { |a, b| a.name <=> b.name }
```

Comment:
Yeah, there is some complexity/oddness about the UX to this, so I'm just kind of making maps masquerade as topics (which contain a link/act as a reference to a map) in the dropdown. Otherwise, you'd be seeing topics with type metamap, and then also maps with type metamap, and prbly get confused. I think this way is ok for now, it feels simpler and intuitive enough on the UI